[gf] Compilation failure with clang-3.4.2

Detected by Jenkins CI build on Pauli, confirmed manually, and is likely the same as the MacPorts build failure https://github.com/ALPSCore/packaging/issues/9

```diff
 Building CXX object utilities/test/CMakeFiles/tensor_test.dir/tensor_test.cpp.o
 cd /home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/build.tmp/utilities/test && /usr/bin/clang++   -isystem /home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/common/deps/gtest-1.7.0/include -I/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/include -I/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/build.tmp/utilities/include -isystem /opt/ohpc/pub/libs/gnu/openmpi/boost/1.66.0/include -I/home/galexv/.local/packages/eigen-3.3.4  -ftemplate-depth=1024 -O3 -DNDEBUG -DBOOST_DISABLE_ASSERTS   -std=c++11 -o CMakeFiles/tensor_test.dir/tensor_test.cpp.o -c /home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/test/tensor_test.cpp
-/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/test/tensor_test.cpp:198:39: warning: too many braces around scalar initializer
-  tensor_view<double, 1> Y(X.data(), {{X.size()}});
                                       ^~~~~~~~~~
-In file included from /home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/test/tensor_test.cpp:10:
-/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/include/alps/numeric/tensors/tensor_base.hpp:173:82: warning: too many braces around scalar initializer
-        tensor_base(T *data, size_t size1, Indices...sizes) : tensor_base(data, {{size1, size_t(sizes)...}}) {}
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
-/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/test/tensor_test.cpp:198:26: note: in instantiation of function template specialization 'alps::numerics::detail::tensor_base<double, 1, alps::numerics::detail::data_view<double> >::tensor_base<>' requested here
-  tensor_view<double, 1> Y(X.data(), {{X.size()}});
                          ^
-In file included from /home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/test/tensor_test.cpp:10:
-/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/utilities/include/alps/numeric/tensors/tensor_base.hpp:173:63: error: constructor for 'tensor_base<>' creates a delegation cycle [-Wdelegating-ctor-cycles]
-        tensor_base(T *data, size_t size1, Indices...sizes) : tensor_base(data, {{size1, size_t(sizes)...}}) {}
                                                               ^
-2 warnings and 1 error generated.
 make[2]: *** [utilities/test/CMakeFiles/tensor_test.dir/tensor_test.cpp.o] Error 1
 make[2]: Leaving directory `/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/build.tmp'
 make[1]: *** [utilities/test/CMakeFiles/tensor_test.dir/all] Error 2
 make[1]: Leaving directory `/home/galexv/.jenkins/workspace/Public_projects/ALPSCore/COMPILER/clang_3.4.2/ENABLE_MPI/OFF/build.tmp'
-make: *** [all] Error 2
```

It is very likely a clang problem, but nevertheless either workaround should be coded, if possible, or the compiler should be blacklisted (and the [prerequisites](https://github.com/ALPSCore/ALPSCore/wiki/Installation#prerequisites) changed).

--- a/utilities/include/alps/numeric/tensors/tensor_base.hpp
+++ b/utilities/include/alps/numeric/tensors/tensor_base.hpp
@@ -170,7 +170,7 @@ namespace alps {
         }
 
         template<typename...Indices>
-        tensor_base(T *data, size_t size1, Indices...sizes) : tensor_base(data, {{size1, size_t(sizes)...}}) {}
+        tensor_base(T *data, size_t size1, Indices...sizes) : storage_(data, size({{size1, size_t(sizes)...}})), shape_({{size1, size_t(sizes)...}}) {fill_acc_sizes();}
 
         /**
          * Create empty Tensor with provided sizes for each dimensions

--- a/utilities/test/tensor_test.cpp
+++ b/utilities/test/tensor_test.cpp
@@ -195,8 +195,8 @@ TEST(TensorTest, TestMultByScalar) {
 TEST(TensorTest, RemoteDataRef) {
   size_t n = 256;
   std::vector<double> X(n, 0.0);
-  tensor_view<double, 1> Y(X.data(), {{X.size()}});
-  tensor_view<double, 2> Z(X.data(), {{16, 16}});
+  tensor_view<double, 1> Y(X.data(), X.size());
+  tensor_view<double, 2> Z(X.data(), 16, 16);
   for(size_t i = 0; i< X.size(); ++i) {
     X[i] = i*0.5;
   }

